pkg/kubelet/container/helpers.go

```diff
@@ -86,10 +86,17 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 	if status.State == ContainerStateRunning {
 		return false
 	}
-	// Always restart container in the unknown, or in the created state.
-	if status.State == ContainerStateUnknown || status.State == ContainerStateCreated {
+	// Always restart container in the unknown state
+	if status.State == ContainerStateUnknown {
 		return true
 	}
+	if status.State == ContainerStateCreated {
```

Comment:
Although it is likely that Generic PLEG might also have the same issue with `ShouldContainerBeRestarted` function, but maybe it is not getting surfaced. But considering it is somehow working with Generic PLEG as is, do you want to surround this change as well with `pleg.IsEventedPLEGInUse()` so it would not change code flow for existing Generic PLEG? 

This way the PR we could be merged as it is not modifying the existing behavior under Generic PLEG and help us  unblock the periodic blocking alpha features job. 

We can leave a comment here that, before Evented PLEG is graduated back to Beta, we should have a concrete answer on why the `ShouldContainerBeRestarted()` is reached when using Evented PLEG but not when using Generic PLEG. 

What do you think?

Reply:
Agree. 

However to avoid import cycle, I add logic when calling `ShouldContainerBeRestarted`. https://github.com/kubernetes/kubernetes/pull/122778/files#diff-0e1a8d22b0f1dc42d412749c6a916b4c76394ead739d4bfbf26564faedd74a2fR971 if evented PLEG is in use, set restartOnCreatedState to false and if not set it true.